regression/utils.py

```diff
@@ -204,3 +205,65 @@ def delete_oci(remote_dir: str) -> None:
 
     for filenames in objects.data.objects:
         oci_client.delete_object(namespace, obj.netloc, filenames.name)
+
+
+def compare_files(files: list[str], identical: bool = False) -> None:
```

Comment:
can we use the library difflib for this? gets the diff between two files. that's what we need right? Example here:
[](https://www.tutorialspoint.com/How-to-find-difference-between-2-files-in-Python#:~:text=We%20invoke%20the%20open(),file%20names%20in%20the%20output.)

Reply:
So the method `compare_files()` is not just the diff comparison, it is actually a sample order comparison between multiple files across all GPUs. For example, each line in an output file contains a sample id of `batch_size * n_gpus_across_all_nodes`  samples. For deterministic sample ordering, given same NCN and global batch_size, each gpu gets some sample ordering, so with one forward pass across all the gpus, the set of sample ids between any two runs should be same, however, the samples can be in different order (in different gpu or in different position in a batch size) in the same forward pass.